src/table/table-header.component.ts

```diff
@@ -0,0 +1,234 @@
+							'bx--table-sort--active': column.sorted,
+							'bx--table-sort--ascending': column.ascending
+						}"
+						(click)="sort.emit(i)">
+						<span
+							*ngIf="!column.template"
+							[title]="column.data"
+							tabindex="-1">
+							{{column.data}}
+						</span>
+						<ng-template
+							[ngTemplateOutlet]="column.template" [ngTemplateOutletContext]="{data: column.data}">
+						</ng-template>
+						<svg
+							focusable="false"
+							preserveAspectRatio="xMidYMid meet"
+							style="will-change: transform;"
+							xmlns="http://www.w3.org/2000/svg"
+							class="bx--table-sort__icon"
+							width="16"
+							height="16"
+							viewBox="0 0 16 16"
+							aria-hidden="true">
+							<path d="M12.3 9.3l-3.8 3.8V1h-1v12.1L3.7 9.3 3 10l5 5 5-5z"></path>
+						</svg>
+						<svg
+							focusable="false"
+							preserveAspectRatio="xMidYMid meet"
+							style="will-change: transform;"
+							xmlns="http://www.w3.org/2000/svg"
+							class="bx--table-sort__icon-unsorted"
+							width="16"
+							height="16"
+							viewBox="0 0 16 16"
+							aria-hidden="true">
+							<path d="M13.8 10.3L12 12.1V2h-1v10.1l-1.8-1.8-.7.7 3 3 3-3zM4.5 2l-3 3 .7.7L4 3.9V14h1V3.9l1.8 1.8.7-.7z"></path>
+						</svg>
+					</button>
+					<span
+						class="bx--table-header-label"
+						*ngIf="!skeleton && this.sort.observers.length === 0 || (this.sort.observers.length > 0 && !column.sortable)">
+						<span *ngIf="!column.template" [title]="column.data">{{column.data}}</span>
+						<ng-template
+							[ngTemplateOutlet]="column.template" [ngTemplateOutletContext]="{data: column.data}">
+						</ng-template>
+					</span>
+					<button
+						[ngClass]="{'active': column.filterCount > 0}"
+						*ngIf="column.filterTemplate"
+						type="button"
+						aria-expanded="false"
+						aria-haspopup="true"
+						[ibmTooltip]="column.filterTemplate"
+						trigger="click"
+						[title]="filterTitle | async"
+						placement="bottom,top"
+						[data]="column.filterData">
+						<svg
+							xmlns="http://www.w3.org/2000/svg"
+							class="icon--sm"
+							width="16"
+							height="16"
+							viewBox="0 0 16 16">
+							<path d="M0 0v3l6 8v5h4v-5l6-8V0H0zm9 10.7V15H7v-4.3L1.3 3h13.5L9 10.7z"/>
+						</svg>
+						<span *ngIf="column.filterCount > 0">
+							{{column.filterCount}}
+						</span>
+					</button>
+				</th>
+			</ng-container>
+			<th *ngIf="!skeleton && stickyHeader" [ngStyle]="{'width': scrollbarWidth + 'px', 'padding': 0, 'border': 0}">
+				<!--
+					Scrollbar pushes body to the left so this header column is added to push
+					the title bar the same amount and keep the header and body columns aligned.
+				-->
+			</th>
+		</tr>
+	`
+})
+export class TableHeader {
+	@Input() model: TableModel;
+
+	@Input() showSelectionColumn: boolean;
+
+	@Input() selectAllCheckboxSomeSelected: boolean;
+
+	@Input() selectAllCheckbox: boolean;
+
+	@Input() skeleton: boolean;
+
+	@Input() set columnIndex(value: number) {
```

Comment:
Just a note, no action required: Since we're splitting this up further we'll want to look at centralizing this in some sort of model/service thing we can share in a given table instance